dev/ScrollViewer/ScrollViewer.cpp

```diff
@@ -1451,86 +1557,96 @@ void ScrollViewer::UpdateScrollControllersVisibility(
     }
 }
 
-bool ScrollViewer::IsLoaded()
-{
-    return winrt::VisualTreeHelper::GetParent(*this) != nullptr;
-}
-
 bool ScrollViewer::IsInputKindIgnored(winrt::InputKind const& inputKind)
 {
     return (IgnoredInputKind() & inputKind) == inputKind;
 }
 
-bool ScrollViewer::AreAllScrollControllersCollapsed()
+bool ScrollViewer::AreAllScrollControllersCollapsed() const
 {
     return (!m_horizontalScrollControllerElement || m_horizontalScrollControllerElement.get().Visibility() == winrt::Visibility::Collapsed) &&
            (!m_verticalScrollControllerElement || m_verticalScrollControllerElement.get().Visibility() == winrt::Visibility::Collapsed);
 }
 
-bool ScrollViewer::AreBothScrollControllersVisible()
+bool ScrollViewer::AreBothScrollControllersVisible() const
 {
     return m_horizontalScrollControllerElement && m_horizontalScrollControllerElement.get().Visibility() == winrt::Visibility::Visible &&
            m_verticalScrollControllerElement && m_verticalScrollControllerElement.get().Visibility() == winrt::Visibility::Visible;
 }
 
-// Show the appropriate scrolling indicators.
-void ScrollViewer::ShowIndicators()
+bool ScrollViewer::AreScrollControllersAutoHiding()
 {
-    if (!AreAllScrollControllersCollapsed())
+    // Use the cached value unless it was invalidated.
+    if (m_autoHideScrollControllersValid)
     {
-        if (m_hideIndicatorsTimer)
-        {
-            winrt::DispatcherTimer hideIndicatorsTimer = m_hideIndicatorsTimer.get();
+        return m_autoHideScrollControllers;
+    }
 
-            if (hideIndicatorsTimer.IsEnabled())
-            {
-                hideIndicatorsTimer.Stop();
-                hideIndicatorsTimer.Start();
-            }
-        }
+    m_autoHideScrollControllersValid = true;
```

Comment:
Odd that a method that sounds like a property (AreX) has side effects. Method name should probably more clearly indicate that.

Reply:
I guess it's a cache value, so maybe it's fine? Anyway, it felt weird.

Reply:
I think it's fine. The property value is updated on demand.